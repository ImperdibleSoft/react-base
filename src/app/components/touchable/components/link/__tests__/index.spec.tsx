import * as React from 'react';
import { shallow } from 'enzyme';

import Link from '..';

describe('Link', () => {
  it('should render a button that looks like a link', () => {
    const component = shallow(<Link options={{ id: 'button', label: 'Link', onClick: () => {} }} />);
    expect(component).toMatchSnapshot();
  });

  it('should render a link', () => {
    const component = shallow(<Link options={{ id: 'button', label: 'Link', to: '/url' }} />);
    expect(component).toMatchSnapshot();
  });

  it('should render a link with a text-icon', () => {
    const component = shallow(
      <Link options={{ id: 'button', label: 'Link', icon: 'send', iconLast: true, to: '/url' }} />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render a react-based link', () => {
    const component = shallow(<Link options={{ id: 'button', label: 'Link', to: { pathname: '/url' } }} />);
    expect(component).toMatchSnapshot();
  });

  it('should render an external link', () => {
    const component = shallow(<Link options={{ id: 'button', isExternal: true, label: 'Link', to: '/url' }} />);
    expect(component).toMatchSnapshot();
  });

  it('should render an invalid link', () => {
    const component = shallow(<Link options={{ id: 'button', label: 'Link' }} />);
    expect(component).toMatchSnapshot();
  });
});
