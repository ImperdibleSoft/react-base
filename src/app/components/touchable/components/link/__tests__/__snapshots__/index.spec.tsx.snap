// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render a button that looks like a link 1`] = `
<ButtonAsLink
  id="button"
  onClick={[Function]}
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Link",
      }
    }
  />
</ButtonAsLink>
`;

exports[`Link should render a link 1`] = `
<LinkAsLink>
  <Link
    id="button"
    onClick={[Function]}
    to={
      Object {
        "pathname": "/url",
        "state": Object {
          "from": "/",
        },
      }
    }
  >
    <Content
      options={
        Object {
          "icon": undefined,
          "iconLast": undefined,
          "label": "Link",
        }
      }
    />
  </Link>
</LinkAsLink>
`;

exports[`Link should render a link with a text-icon 1`] = `
<LinkAsLink>
  <Link
    id="button"
    onClick={[Function]}
    to={
      Object {
        "pathname": "/url",
        "state": Object {
          "from": "/",
        },
      }
    }
  >
    <Content
      options={
        Object {
          "icon": "send",
          "iconLast": true,
          "label": "Link",
        }
      }
    />
  </Link>
</LinkAsLink>
`;

exports[`Link should render a react-based link 1`] = `
<LinkAsLink>
  <Link
    id="button"
    onClick={[Function]}
    to={
      Object {
        "pathname": "/url",
      }
    }
  >
    <Content
      options={
        Object {
          "icon": undefined,
          "iconLast": undefined,
          "label": "Link",
        }
      }
    />
  </Link>
</LinkAsLink>
`;

exports[`Link should render an external link 1`] = `
<LinkAsLink>
  <a
    href="/url"
    id="button"
    onClick={[Function]}
    rel="noopener noreferrer"
    target="_blank"
  >
    <Content
      options={
        Object {
          "icon": undefined,
          "iconLast": undefined,
          "label": "Link",
        }
      }
    />
  </a>
</LinkAsLink>
`;

exports[`Link should render an invalid link 1`] = `
<ButtonAsLink
  id="button"
  onClick={[Function]}
>
  <Content
    options={
      Object {
        "icon": undefined,
        "iconLast": undefined,
        "label": "Link",
      }
    }
  />
</ButtonAsLink>
`;
